### PR TITLE
fix(ExpectedConditions): non-static ExpectedConditions for browser

### LINKED ISSUE #3761: Issue with ExpectedConditions working with forked browser instances
Bug report
---

- Node Version: `v6.9.1`
- Protractor Version: `4.0.11`
- Angular Version: `1.5.6`
- Browser(s): `chrome`
- Operating System and Version `mac osx`

Code of my test is as following:

```

beforeAll(function(){
  browser2=browser.forkNewDriverInstance();
  return user.login(browser2);
});

beforeEach(function(){
  return browser2.get('/other-page');
});
```

and user.login() is:

```
User.prototype.login=function(browserToUse){
  browser.runLogin()
    .then(function(){
      return browserToUse.wait(EC.urlContains('/home'), 5000*2,'Waiting ofr URL /home');
    });
}
```

It looks like that it executes instructions in the wrong order, that is:

1. user.login(b)
2. browser2.get('/other-page')
3. browserToUse.wait(EC.urlContains('/home'), 5000*2,'Waiting ofr URL /home');

indeed, the 3rd instruction generates a time out exception. But the browser goes through /home URL, it is a problem of synchronization.

I wonder: how is it possible? 

Which is the way to tell protractor: don't use your algorithm for ordering instructions, but I manage the order?

I am using `.then()` for each instruction, but it does not seem enough.

It is worth to note that:
* this test (that includes the beforeAll) have other tests (describe and it) and that before it
* when i run this test by itself, it does not go in timeout
* when i run the whole file, it goes in timeout

### PR DESCRIPTION
**Pull request by cnishina**
closes #3761

### CODE REVIEW

**Comment by cnishina** [comment]
It looks like:

* The basic tests pass locally and on sauce.
* when Protractor shuts down, `webdriver.quit` throws an error.
* After some investigation, forking a browser instance is causing an error only in sauce labs with the `spec/basic/expected_conditions_spec.js`:

```
var browser2 = browser.forkNewDriverInstance();
browser2.get('index.html#/form');
var EC2 = browser2.ExpectedConditions;
```

My guess is that we are not shutting down properly and the issue might be in `lib/driverProviders/driverProviders.ts` and `lib/driverProviders/sauce.ts`...

**Comment by cnishina** [comment]
Also able to reproduce with `driverProviders/local` :confused:

**Comment by cnishina** [comment]
In a related issue https://github.com/angular/protractor/issues/3772

File: lib/browser.ts [hunk 1]
```diff
@@ -320,6 +320,9 @@ export class ProtractorBrowser extends Webdriver {
     this.trackOutstandingTimeouts_ = !opt_untrackOutstandingTimeouts;
     this.mockModules_ = [];
     this.addBaseMockModules_();
+
+    // set up expected conditions
```
Comment (juliemr) [inline lib/browser.ts]: Nit: comments should be sentences.

Comment (cnishina) [inline lib/browser.ts]: :disappointed: Will fix in next PR...

**Pull request merged**